packages/core/src/app/address/AddressForm.tsx

```diff
@@ -1,214 +1,89 @@
-    onAutocompleteToggle?(state: { inputValue: string; isOpen: boolean }): void;
-    onChange?(fieldName: string, value: string | string[]): void;
-    setFieldValue?(fieldName: string, value: string | string[]): void;
-}
-
-const LABEL: AddressKeyMap = {
-    address1: 'address.address_line_1_label',
-    address2: 'address.address_line_2_label',
-    city: 'address.city_label',
-    company: 'address.company_name_label',
-    countryCode: 'address.country_label',
-    firstName: 'address.first_name_label',
-    lastName: 'address.last_name_label',
-    phone: 'address.phone_number_label',
-    postalCode: 'address.postal_code_label',
-    stateOrProvince: 'address.state_label',
-    stateOrProvinceCode: 'address.state_label',
-};
-
-const AUTOCOMPLETE: AddressKeyMap = {
-    address1: 'address-line1',
-    address2: 'address-line2',
-    city: 'address-level2',
-    company: 'organization',
-    countryCode: 'country',
-    firstName: 'given-name',
-    lastName: 'family-name',
-    phone: 'tel',
-    postalCode: 'postal-code',
-    stateOrProvince: 'address-level1',
-    stateOrProvinceCode: 'address-level1',
-};
-
-const PLACEHOLDER: AddressKeyMap = {
-    countryCode: 'address.select_country_action',
-    stateOrProvince: 'address.select_state_action',
-    stateOrProvinceCode: 'address.select_state_action',
-};
-
-const AUTOCOMPLETE_FIELD_NAME = 'address1';
+import './AddressForm.scss'; 
 
-class AddressForm extends Component<AddressFormProps & WithLanguageProps> {
-    private containerRef: RefObject<HTMLElement> = createRef();
-    private nextElement?: HTMLElement | null;
-    static contextType = ThemeContext;
-    declare context: React.ContextType<typeof ThemeContext>;
-
-    private handleDynamicFormFieldChange: (name: string) => (value: string | string[]) => void =
-        memoize((name) => (value) => {
-            this.syncNonFormikValue(name, value);
-        });
-
-    componentDidMount(): void {
-        const { current } = this.containerRef;
+const AddressForm: React.FC<AddressFormProps> = ({
+        formFields,
+        fieldName,
+        countryCode,
+        onAutocompleteToggle,
+        shouldShowSaveAddress,
+        setFieldValue = noop,
+        onChange = noop,
+        type,
+    }) => {
+    const { language } = useLocale();
+    const { themeV2 } = useThemeContext();
+    const {
+        checkoutState: {
+            data: { getConfig, getBillingCountries, getShippingCountries }
+        }
+    } = useCheckout();
+    
+    const config = getConfig();
+    const countries = (type === AddressType.Billing 
+        ? getBillingCountries() 
+        : getShippingCountries()) 
+        || EMPTY_ARRAY;
+        const googleMapsApiKey = config?.checkoutSettings.googleMapsApiKey || '';
+    const isFloatingLabelEnabledValue = config ? isFloatingLabelEnabled(config.checkoutSettings) : false;
+    const countriesWithAutocomplete = ['US', 'CA', 'AU', 'NZ', 'GB'];
```

Comment:
`const countriesWithAutocomplete = ['US', 'CA', 'AU', 'NZ', 'GB']` has a same copy in `BillingForm.tsx`, can we consolidate them into one place one copy?